### PR TITLE
Improve step 8

### PR DESCRIPTION
**Pull request by drwells**
I have a student who is going to do a little bit of work on solid mechanics and I pointed him at this step. I went ahead and improved a few small things:
1. We now set up constraints in the same way we do in step-6.
2. I removed `<code></code>` so that we get more doxygen links.
3. We don't need a destructor.

### CODE REVIEW

File: examples/step-8/step-8.cc [hunk 1]
```diff
@@ -598,13 +559,12 @@ namespace Step8
...
-  // <code>triangulation</code> object will not flag any cells for refinement
-  // (why should it if the indicated error is zero for each cell?). The grid
-  // in the next iteration will therefore consist of four cells only as well,
-  // and the same problem occurs again.
+  // the error indicators computed by the KellyErrorEstimator class are zero
+  // for each cell as well, and the call to
+  // Triangulation::refine_and_coarsen_fixed_number() on the will not flag any
+  // cells for refinement (why should it if the indicated error is zero for
+  // each cell?). The grid in the next iteration will therefore consist of
+  // four cells only as well, and the same problem occurs again.
```
Comment (tjhei) [inline examples/step-8/step-8.cc:567]: Can you please fix the "twice" in the paragraph below as well?

File: examples/step-8/step-8.cc [hunk 2]
```diff
@@ -598,13 +559,12 @@ namespace Step8
...
-  // the error indicators computed by the <code>KellyErrorEstimator</code>
-  // class are zero for each cell as well, and the call to
-  // <code>refine_and_coarsen_fixed_number</code> on the
-  // <code>triangulation</code> object will not flag any cells for refinement
-  // (why should it if the indicated error is zero for each cell?). The grid
-  // in the next iteration will therefore consist of four cells only as well,
-  // and the same problem occurs again.
+  // the error indicators computed by the KellyErrorEstimator class are zero
+  // for each cell as well, and the call to
+  // Triangulation::refine_and_coarsen_fixed_number() on the will not flag any
```
Comment (tjhei) [inline examples/step-8/step-8.cc:564]: On the will? 😀

Comment (drwells) [reply examples/step-8/step-8.cc:564]: whoops - I only got partway through 'on the triangulation'. Let me fix that.

Comment (drwells) [reply examples/step-8/step-8.cc]: Fixed.

File: examples/step-8/step-8.cc (hunk 1 above)
Comment (drwells) [reply examples/step-8/step-8.cc]: It's been a long week. Where is the issue here?

Comment (drwells) [reply examples/step-8/step-8.cc]: oh, we refine four times.

Comment (drwells) [reply examples/step-8/step-8.cc]: Also fixed!

**Pull request merged**